### DISCUSSION TITLE
Reading http request headers

### DISCUSSION THREAD

**Discussion by NikaTskhadaia** [canton7/RestEase#210] [Q&A]
I 'm using the library in a gateway to pass incoming HTTP request to different APIs.

Is there any built-in solution for passing incoming request headers in the gateway to outgoing request headers?

**Comment by canton7** [comment]
The various ways of passing headers are [described here](https://github.com/canton7/RestEase#headers). If your `[Body]` parameter is a `HttpContent`, its content headers will also be respected.

That said, is there a reason you're using RestEase for this? If you're just proxying requests, it seems to be that you'll probably have an easier time working directly with `HttpClient`?
